src/Raku/ast/signature.rakumod

```diff
@@ -111,6 +111,27 @@ class RakuAST::Signature
                         }
                         %seen{$name} := 1;
                     }
+
+                    unless $_.is-declared-required || nqp::eqaddr($_.type, RakuAST::Type) {
+                        my $param-type := $_.type.meta-object;
+                        my $param-default := $_.default;
+                        if nqp::istype($param-type.HOW, Perl6::Metamodel::SubsetHOW) {
+                            if ! nqp::eqaddr($param-default, RakuAST::Expression) {
+                                if nqp::can($param-default, 'maybe-compile-time-value')
```

Comment:
The proper check is $param-default.has-compile-time-value. If that returns true, the object must support maybe-compile-time-value.